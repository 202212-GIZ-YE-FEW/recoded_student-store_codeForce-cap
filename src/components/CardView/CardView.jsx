import Image from "next/image"

function ImageList(personalURL, imageName, altName) {
  return (
    <li>
      <a href={personalURL} className='text-gray-500 hover:text-gray-900'>
        <Image src={imageName} width={35} height={35} alt={altName} />
      </a>
    </li>
  )
}

function CardView(props) {
  const images = {
    facebook: "facebook-circle.svg",
    twitter: "twitter-circle.svg",
    github: "github.svg",
  }

  return (
    <div className='items-center bg-gray-50 rounded-lg shadow sm:flex'>
      <Image
        className='w-full rounded-lg sm:rounded-none sm:rounded-l-lg'
        src={props.profile}
        alt='profile'
        width={100}
        height={100}
      />

      <div className='p-5'>
        <h3 className='text-xl font-bold tracking-tight text-black'>
          {props.fullName}
        </h3>

        <span className='text-gray-500'>{props.jobName}</span>

        <p className='mt-3 mb-4 font-semibold text-gray-500'>
          {props.description}
        </p>

        <ul className='flex space-x-4 sm:mt-0'>
          {ImageList("#", images.facebook, "facebook")}
          {ImageList("#", images.twitter, "twitter")}
          {ImageList("#", images.github, "github")}
        </ul>
      </div>
    </div>
  )
}

export default CardView
