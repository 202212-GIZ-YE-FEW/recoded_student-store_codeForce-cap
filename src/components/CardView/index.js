import CardView from "./CardView"

export default CardView
