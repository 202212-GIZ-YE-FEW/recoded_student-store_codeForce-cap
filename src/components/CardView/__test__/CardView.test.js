import renderer from "react-test-renderer"

import CardView from ".."

it("renders correctly", () => {
  const tree = renderer.create(<CardView />).toJSON()
  expect(tree).toMatchSnapshot()
})
