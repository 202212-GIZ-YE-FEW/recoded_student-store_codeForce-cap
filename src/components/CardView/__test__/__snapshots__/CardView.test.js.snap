// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="items-center bg-gray-50 rounded-lg shadow sm:flex"
>
  <a
    href="#"
  >
    <img
      alt=""
      className="w-full rounded-lg sm:rounded-none sm:rounded-l-lg"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="cat-photo.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={100}
    />
  </a>
  <div
    className="p-5"
  >
    <h3
      className="text-xl font-bold tracking-tight text-black"
    >
      Full Name
    </h3>
    <span
      className="text-gray-500"
    >
      Web Developer
    </span>
    <p
      className="mt-3 mb-4 font-semibold text-gray-500"
    >
      describe
    </p>
    <ul
      className="flex space-x-4 sm:mt-0"
    >
      <li>
        <a
          className="text-gray-500 hover:text-gray-900"
          href="#"
        >
          <img
            alt="facebook"
            data-nimg="1"
            decoding="async"
            height={35}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="facebook-circle.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={35}
          />
        </a>
      </li>
      <li>
        <a
          className="text-gray-500 hover:text-gray-900"
          href="#"
        >
          <img
            alt="twitter"
            data-nimg="1"
            decoding="async"
            height={35}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="twitter-circle.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={35}
          />
        </a>
      </li>
      <li>
        <a
          className="text-gray-500 hover:text-gray-900"
          href="#"
        >
          <img
            alt="github"
            data-nimg="1"
            decoding="async"
            height={35}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="github.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={35}
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
